R/tinyplot.R

```diff
@@ -519,6 +529,32 @@ tinyplot.default = function(
 
   xlabs = NULL
 
+  ## handle defaults of axes, xaxt, yaxt, frame.plot
+  ## - convert axes to character if necessary
+  ## - set defaults of xaxt/yaxt if these are NULL
+  ## - set logical axes based on xaxt/yaxt
+  ## - set frame.plot default based on xaxt/yaxt
+  if (!is.character(axes)) axes = if (isFALSE(axes)) "none" else "standard"
+  axis_types = c("standard", "none", "labels", "ticks", "axis")
+  axes = match.arg(axes, axis_types)
+  if (is.null(xaxt)) xaxt = axes
+  if (is.null(yaxt)) yaxt = axes
+  xaxt = substr(match.arg(xaxt, axis_types), 1L, 1L)
+  yaxt = substr(match.arg(yaxt, axis_types), 1L, 1L)
+  axes = any(c(xaxt, yaxt) != "n")
+  if (is.null(frame.plot) || !is.logical(frame.plot)) frame.plot = all(c(xaxt, yaxt) %in% c("s", "a"))
+
+  ## auxiliary function with axis type
```

Comment:
My vote is to move to its own file as an auxiliary function.

Reply:
Sure, will do. Any name preferences?

- [ ] `tinyAxis()`
- [ ] `tiny_Axis()`
- [ ] `tiny_axis()`
- [ ] Other

Reply:
I like `tinyAxis` for consistency.

Reply:
:+1: